packages/ui/components/ui/account-switcher.tsx

```diff
@@ -13,42 +13,85 @@ const MAX_INDEX = 2 ** 32;
 export const AccountSwitcher = ({
   account,
   onChange,
+  filter,
 }: {
   account: number;
   onChange: (account: number) => void;
+  /**
+   * An array of address indexes to switch between, if you want to limit the
+   * options. Must be pre-sorted! `AccountSwitcher` won't sort this for you.
+   */
+  filter?: number[];
 }) => {
   const [inputCharWidth, setInputCharWidth] = useState(1);
 
+  const sortedFilter = useMemo(() => (filter ? [...filter].sort() : undefined), [filter]);
+
+  const handleClickPrevious = () => {
+    if (sortedFilter) {
+      const previousAccount = sortedFilter[sortedFilter.indexOf(account) - 1];
+      if (previousAccount !== undefined) onChange(previousAccount);
+    } else onChange(account - 1);
+  };
+
+  const handleClickNext = () => {
+    if (sortedFilter) {
+      const nextAccount = sortedFilter[sortedFilter.indexOf(account) + 1];
+      if (nextAccount !== undefined) onChange(nextAccount);
+    } else onChange(account + 1);
+  };
+
+  const shouldShowPreviousButton =
+    account !== 0 && (!sortedFilter || sortedFilter.indexOf(account) > 0);
+  const shouldShowNextButton =
+    account !== MAX_INDEX &&
+    (!sortedFilter || sortedFilter.indexOf(account) < sortedFilter.length - 1);
+
   const handleChange = (value: number) => {
     onChange(value);
     setInputCharWidth(String(value).length);
   };
 
   return (
     <div className='flex items-center justify-between'>
-      <Button variant='ghost' className={cn('hover:bg-inherit', account === 0 && 'cursor-default')}>
-        {account !== 0 ? (
+      {shouldShowPreviousButton ? (
+        <Button
+          variant='ghost'
+          className={cn('hover:bg-inherit', account === 0 && 'cursor-default')}
+        >
           <ArrowLeftIcon
-            onClick={() => {
-              if (account > 0) handleChange(account - 1);
-            }}
+            aria-label='Previous account'
+            role='button'
+            onClick={handleClickPrevious}
             className='size-6 hover:cursor-pointer'
           />
-        ) : (
-          <span className='size-6' />
-        )}
-      </Button>
+        </Button>
+      ) : (
+        <span className='size-6' />
+      )}
```

Comment:
I also noticed here that we weren't hiding the `Button` element, but only the arrow element — meaning that, when we shouldn't have the Previous button at all, we were still rendering an invisible one 😆 

Same for the Next button at the bottom of this file.